lib/cli/interactive-setup/service.js

```diff
@@ -55,27 +58,69 @@ const projectNameInput = async (workingDir) =>
     })
   ).projectName.trim();
 
+const resolveProjectNameInput = async (options, workingDir) => {
+  if (options.name) {
+    if (!isValidServiceName(options.name)) {
+      throw new ServerlessError(INVALID_PROJECT_NAME_MESSAGE, 'INVALID_PROJECT_NAME');
+    }
+
+    let alreadyTaken = false;
+    try {
+      await fs.promises.access(join(workingDir, options.name));
+      alreadyTaken = true;
+    } catch {
+      // Pass
+    }
+
+    if (alreadyTaken) {
+      throw new ServerlessError(
+        `Path ${options.name} is already taken`,
+        'TARGET_FOLDER_ALREADY_EXISTS'
+      );
+    }
+
+    return options.name;
+  }
+
+  return projectNameInput(workingDir);
+};
+
 module.exports = {
   isApplicable({ serviceDir }) {
     return !serviceDir;
   },
   async run(context) {
     const workingDir = context.cwd || process.cwd();
+    const options = context.options || {};
     const isConfirmed = await confirm('No project detected. Do you want to create a new one?', {
       name: 'shouldCreateNewProject',
     });
     if (!isConfirmed) return;
-    const projectType = await projectTypeChoice();
-    if (projectType === 'other') {
-      process.stdout.write(
-        '\nRun “serverless create --help” to view available templates and create a new project ' +
-          'from one of those templates.\n'
-      );
-      return;
+
+    let projectDir;
+    let projectName;
+    if (options['template-path']) {
+      projectName = await resolveProjectNameInput(options, workingDir);
+      projectDir = join(workingDir, projectName);
+      createFromLocalTemplate({
```

Comment:
I've went with the approach of throwing errors in cases where the invalid options were supplied (`name` and `template-path`), but I'm not sure about the approach

Reply:
IMO good call, I think we should validate CLI options with real thrown errors, as those are forced settings (we're not re-asking user to pass fixed CLI option), and via CLI options we also want to support non interactive usage, and in such scenario it's very important that processes exit with non zero code.